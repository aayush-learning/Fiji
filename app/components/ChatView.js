import React, { Component } from 'react';
import { Text, View, StyleSheet, Image, KeyboardAvoidingView, ListView, TextInput } from 'react-native';
import PropTypes from 'prop-types'
import SvgUri from 'react-native-svg-uri'
import { sendMessage, startChat } from '../redux/chat';


class ChatView extends Component {
  
  render() {

    console.log("this is sender::" + this.props.item.sender)
    console.log("this is user::" + this.props.friend._id)
    const Eachmsg = (props) => {
      //show message from sender
      if (this.props.item.sender == this.props.friend._id) 
        return (
          <View style={styles.eachMsg}>
            <Image source={{ uri: 'data:image/png;base64,' + this.props.friend.image }} style={styles.userPic} />
              <View style={styles.msgBlock}>
              <Text style={styles.msgTxt}>{this.props.item.text}</Text>
            </View>
          </View>
        );

        return (
          <View style={styles.rightMsg} >
            <View style={styles.rightBlock} >
              <Text style={styles.rightTxt}>{this.props.item.text}</Text>
            </View>
            <Image source={{ uri: 'data:image/png;base64,' + this.props.user.image }} style={styles.userPic} />
          </View>
        );
      
    }
    return (
          <View>
          <Eachmsg />
          </View>
    );

  }
}

ChatView.propTypes = {
  text: PropTypes.object,
  sender: PropTypes.object,
  user: PropTypes.object,
  message: PropTypes.object,
  friend: PropTypes.object,
  image: PropTypes.object
}
export default ChatView;

const styles = StyleSheet.create({
  keyboard: {
    flex: 1,
    justifyContent: 'center',
  },
  image: {
    position: "absolute",
    width: "100%",
    height: "100%",
  },
  header: {
    height: 65,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: '#075e54',
  },
  left: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  right: {
    flexDirection: 'row',
  },
  chatTitle: {
    color: '#fff',
    fontWeight: '600',
    margin: 10,
    fontSize: 15,
  },
  chatImage: {
    width: 30,
    height: 30,
    borderRadius: 15,
    margin: 5,
  },
  input: {
    flexDirection: 'row',
    alignSelf: 'flex-end',
    padding: 10,
    // height: 60,
    width: "90%",
    backgroundColor: '#fff',
    margin: 10,
    borderRadius: 30,
    shadowColor: '#3d3d3d',
    shadowRadius: 2,
    shadowOpacity: 0.5,
    shadowOffset: {
      height: 1,
    },
  },
  eachMsg: {
    flexDirection: 'row',
    alignItems: 'flex-end',
    // backgroundColor:"#19a4f2",
    margin: 5,
  },
  rightMsg: {
    flexDirection: 'row',
    alignItems: 'flex-end',
    // backgroundColor:"#19a4f2",
    margin: 5,
    alignSelf: 'flex-end',
  },
  userPic: {
    height: 40,
    width: 40,
    margin: 5,
    borderRadius: 20,
    // backgroundColor: '#f8f8f8',
  },
  msgBlock: {
    width: 220,
    borderRadius: 5,
    backgroundColor: '#edeeef',
    padding: 10,
    shadowColor: '#adef12',
    shadowRadius: 2,
    shadowOpacity: 0.8,
    shadowOffset: {
      height: 1,
    },
  },
  rightBlock: {
    width: 220,
    borderRadius: 5,
    backgroundColor:"#19a4f2",
    padding: 10,
    shadowColor: '#3d3d3d',
    shadowRadius: 2,
    shadowOpacity: 0.5,
    shadowOffset: {
      height: 1,
    },
  },
  msgTxt: {
    fontSize: 15,
    color: '#555',
    fontWeight: '600',
  },
  rightTxt: {
    fontSize: 15,
    color: '#202020',
    fontWeight: '600',
  },
});
