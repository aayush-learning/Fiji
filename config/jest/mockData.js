import { fetchUsersFailure } from '../../app/redux/users';

const contentJson = require('../seed/content.json')
const usersJson = require('../seed/users.json')
const userAliceJson = require('../seed/user_alice.json')
const storyJson = require('../seed/story.json')

export const titles = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('storytitle:')
})

export const stories = storyJson['docs'].filter(function (doc) {
  return doc._id.startsWith('story:')
})

export const games = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('game:')
})

export const gameThemeData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('gametheme:')
})

export const stickersData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('sticker:caterpillar:')
})

export const stickerPacksData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('sticker-pack:')
})

export const highScoresData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('high-score:')
})

export const story_alice_in_wonderland = storyJson['docs'].filter(function (doc) {
  return doc._id == 'story:alice-in-wonderland'
})[0]

export const users = usersJson['docs']

export const messages = userAliceJson['docs'].filter(function(doc) {
  return doc._id.startsWith('chat:mad-hatter:')
})