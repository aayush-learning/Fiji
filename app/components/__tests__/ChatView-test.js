import 'react-native';
import React from 'react';
import ChatView from '../ChatView';
import { messages, users } from '../../../config/jest/mockData'

import renderer from 'react-test-renderer';



it('renders correctly', () => {
    const tree = renderer.create(
    <ChatView
    item = {messages[0]}
      // to send user,friend to chatview
      // sticker = {this.props.sticker}
     user={users[0]}
       friend={users[1]}  
    />
).toJSON();
    expect(tree).toMatchSnapshot();
  });