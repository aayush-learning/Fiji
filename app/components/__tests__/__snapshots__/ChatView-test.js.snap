// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flexDirection": "row",
        "margin": 5,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADMAAAAzCAYAAAA6oTAqAAAAEXRFWHRTb2Z0d2FyZQBwbmdjcnVzaEB1SfMAAABQSURBVGje7dSxCQBACARB+2/ab8BEeQNhFi6WSYzYLYudDQYGBgYGBgYGBgYGBgYGBgZmcvDqYGBgmhivGQYGBgYGBgYGBgYGBgYGBgbmQw+P/eMrC5UTVAAAAABJRU5ErkJggg==",
        }
      }
      style={
        Object {
          "borderRadius": 20,
          "height": 40,
          "margin": 5,
          "width": 40,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#f2f2f2",
          "borderRadius": 5,
          "padding": 10,
          "shadowColor": "#adef12",
          "shadowOffset": Object {
            "height": 1,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
          "width": 220,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Care for a tea party
      </Text>
    </View>
  </View>
</View>
`;
